src/Web/Areas/Identity/Pages/Account/Login.cshtml.cs

```diff
@@ -99,5 +103,15 @@ public async Task<IActionResult> OnPostAsync(string returnUrl = null)
             // If we got this far, something failed, redisplay form
             return Page();
         }
+
+        private async Task TransferAnonymousBasketToUserAsync(string userName)
```

Comment:
Would you be able to write some tests around this as well? This should also probably be pulled out in to the `BasketService`

Reply:
This method is just calling the TransferBasketAsync method of the basket service in case a cookie with anonymous buyer id exists. I created this private method so that its content wouldn't make the OnPostAsync method fatter.

```csharp
        private async Task TransferAnonymousBasketToUserAsync(string userName)
        {
            if (Request.Cookies.ContainsKey(Constants.BASKET_COOKIENAME))
            {
                var anonymousId = Request.Cookies[Constants.BASKET_COOKIENAME];
                await _basketService.TransferBasketAsync(anonymousId, userName);
                Response.Cookies.Delete(Constants.BASKET_COOKIENAME);
            }
        }
```
This method is private and contains web elements. 

> This should also probably be pulled out in to the BasketService
By this, did you mean that this should be in the ApplicationCore/BasketService instead of Login.cshtml.cs?

Reply:
I see now that it's dealing mostly with Web elements, and calling the existing method on the `BasketService`. I'm ok with how you've done this. No changes required for this.